### PR TITLE
chore: harmonize dependencies

### PR DESCRIPTION
**Pull request by Flarna**
## Short description of the changes

Harmonize dependencies by using following rules:
* peer and dev depend on API ^1.0.0 (like in core repo)
* depend on SDK ^1.0.0 (don't limit user)
* dev-depend on latest SDK pinned (like other deps)
* pin dependencies within the lerna project (like in core repo)

This is a followup of #1012. I decided to create a new PR as it is no longer about API deps only

## Checklist

- [ ] Ran `npm run test-all-versions` for the edited package(s) on the latest commit if applicable.

### CODE REVIEW

File: plugins/node/instrumentation-fs/package.json [hunk 1]
```diff
@@ -42,14 +42,13 @@
...
   "devDependencies": {
-    "@opentelemetry/api": "1.0.4",
+    "@opentelemetry/api": "^1.0.0",
     "@opentelemetry/context-async-hooks": "1.2.0",
     "@opentelemetry/resources": "1.2.0",
     "@opentelemetry/sdk-trace-base": "1.2.0",
     "@opentelemetry/sdk-trace-node": "1.2.0",
     "@types/mocha": "7.0.2",
     "@types/node": "14.17.9",
-    "@types/shimmer": "1.0.2",
```
Comment (Flarna) [inline plugins/node/instrumentation-fs/package.json:52]: seem unused

Comment (dyladan) [reply plugins/node/instrumentation-fs/package.json:52]: shimmer shouldn't be used in any instrumentation directly

Comment (Flarna) [reply plugins/node/instrumentation-fs/package.json:52]: I know, but it turned out that dns uses types only in tests and `@opentelemetry/plugin-react-load` really uses shimmer.

As far as I remember `@opentelemetry/plugin-react-load` is somewhat different as it doesn't depend on `@opentelemetry/instrumentation`. I think it's what the name tells - a plugin for react.

File: packages/opentelemetry-browser-extension-autoinjection/package.json [hunk 2]
```diff
@@ -66,18 +66,18 @@
...
-    "@opentelemetry/exporter-zipkin": "1.2.0",
-    "@opentelemetry/instrumentation": "0.28.0",
-    "@opentelemetry/instrumentation-document-load": "^0.27.1",
+    "@opentelemetry/context-zone": "^1.0.0",
+    "@opentelemetry/core": "^1.0.0",
+    "@opentelemetry/exporter-trace-otlp-http": "^0.28.0",
+    "@opentelemetry/exporter-zipkin": "^1.0.0",
+    "@opentelemetry/instrumentation": "^0.28.0",
+    "@opentelemetry/instrumentation-document-load": "0.27.1",
     "@opentelemetry/instrumentation-fetch": "0.28.0",
```
Comment (blumamir) [inline packages/opentelemetry-browser-extension-autoinjection/package.json:75]: I think this should be with caret as it's a core repo dependency

**Review by blumamir — APPROVED** [review]

**Review by Rauno56 — APPROVED** [review]

**Pull request merged**